qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/project/QuteProjectRegistry.java

```diff
@@ -1480,4 +1506,92 @@ public void dispose() {
 			project.dispose();
 		}
 	}
+
+	public CompletableFuture<ProjectInfo> getProjectInfo(QuteProjectParams params) {
+		return projectInfoProvider.getProjectInfo(params);
+	}
+
+	/**
+	 * Try to load the Qute project of the given workspace folder.
+	 * 
+	 * @param workspaceFolder the workspace folder.
+	 * @param progressSupport the LSP client progress support and null otherwise.
+	 */
+	public void tryToLoadQuteProject(WorkspaceFolder workspaceFolder, ProgressSupport progressSupport) {
+		String projectName = workspaceFolder.getName();
+
+		String progressId = createAndStartProgress(projectName, progressSupport);
+
+		// Load Qute project from the Java component (collect Java data model)
+		String projectUri = workspaceFolder.getUri();
+		QuteProjectParams params = new QuteProjectParams(projectUri);
+		getProjectInfo(params)
+				.thenAccept(projectInfo -> {
+					if (projectInfo == null) {
+						// The workspace folder is not a Qute project, end the process
+						endProgress(progressId, progressSupport);
+						return;
+					}
+
+					// The workspace folder is a Qute project, load the data model from Java
+					// component
+					QuteProject project = getProject(projectInfo, false);
+					if (progressSupport != null) {
+						WorkDoneProgressReport report = new WorkDoneProgressReport();
+						report.setMessage("Loading data model for '" + projectName + "' Qute project.");

```

Comment:
I wonder if we can just put `"Loading '" + projectName + "' Qute project."` to make things simpler